kyo-bench/shared/main/scala/kyo/bench/Bench.scala

```diff
@@ -24,12 +25,24 @@ import zio.UIO
 )
 @BenchmarkMode(Array(Mode.Throughput))
 abstract class Bench[T](val expectedResult: T):
+    private var finalizers: List[() => Unit] = Nil
+
+    @TearDown
+    def tearDown(): Unit = finalizers.foreach(_())
+
+    def zioRuntimeLayer: ZLayer[Any, Any, Any] = ZLayer.empty
 
     lazy val zioRuntime =
         val replaceZioExecutor =
             System.getProperty("replaceZioExecutor", "false") == "true"
+
         if !replaceZioExecutor then
-            zio.Runtime.default.unsafe
+            if zioRuntimeLayer ne ZLayer.empty then
+                val (runtime, finalizer) = ZIORuntime.fromLayerWithFinalizer(zioRuntimeLayer)
+                finalizers = finalizer :: finalizers
+                runtime.unsafe
+            else
+                zio.Runtime.default.unsafe
         else
             KyoSchedulerZioRuntime.default.unsafe
```

Comment:
Would it be possible to support the case when Kyo's scheduler is enabled for ZIO?

Reply:
With the layer mechanism? Definitely

Reply:
I just pushed a change to do so, but I realized that the `override val zioRuntimeLayer = zio.http.Client.default` will hide the fact that the kyo scheduler is not being used. We would need to write:

```
override def zioRuntimeLayer = super.zioRuntimeLayer ++ zio.http.Client.default
```

Reply:
I think it could be a `def` since `zioRuntime` is a `lazy val` and will call `zioRuntimeLayer` only once. Don't you need to handle the `else` branch here when Kyo's scheduler is enabled?

Reply:
You are correct, we need to ensure the kyo scheduler is combined with the layers in the child bench. This is fixed in my latest commit.